service/orchestrator/toe_test.go

```diff
@@ -0,0 +1,362 @@
+	// 1st case: No ToEs stored
+	listTargetsOfEvaluationResponse, err = orchestratorService.ListTargetsOfEvaluation(context.Background(), &orchestrator.ListTargetsOfEvaluationRequest{})
+	assert.NoError(t, err)
+	assert.NotNil(t, listTargetsOfEvaluationResponse.Toes)
+	assert.Empty(t, listTargetsOfEvaluationResponse.Toes)
+
+	// 2nd case: One ToE stored
+	err = orchestratorService.storage.Create(orchestratortest.NewTargetOfEvaluation())
+	assert.NoError(t, err)
+
+	listTargetsOfEvaluationResponse, err = orchestratorService.ListTargetsOfEvaluation(context.Background(), &orchestrator.ListTargetsOfEvaluationRequest{})
+	assert.NoError(t, err)
+	assert.NotNil(t, listTargetsOfEvaluationResponse.Toes)
+	assert.NotEmpty(t, listTargetsOfEvaluationResponse.Toes)
+	assert.Equal(t, 1, len(listTargetsOfEvaluationResponse.Toes))
+
+	// 3rd case: Invalid request
+	_, err = orchestratorService.ListTargetsOfEvaluation(context.Background(),
+		&orchestrator.ListTargetsOfEvaluationRequest{OrderBy: "not a field"})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+	assert.Contains(t, err.Error(), api.ErrInvalidColumnName.Error())
+}
+
+func Test_UpdateTargetOfEvaluation(t *testing.T) {
+	var (
+		toe *orchestrator.TargetOfEvaluation
+		err error
+	)
+	orchestratorService := NewService()
+
+	// 1st case: ToE is nil
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+
+	// 2nd case: Ids are empty
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		CloudServiceId: "",
+		CatalogId:      "",
+	})
+	assert.Equal(t, codes.InvalidArgument, status.Code(err))
+
+	// 3rd case: ToE not found since there are no ToEs
+	_, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		Toe: &orchestrator.TargetOfEvaluation{
+			CloudServiceId: "MyService",
+			CatalogId:      "Cat1234",
+		},
+	})
+	assert.Equal(t, codes.NotFound, status.Code(err))
+
+	// 4th case: ToE updated successfully
+	err = orchestratorService.storage.Create(&orchestrator.TargetOfEvaluation{
+		CloudServiceId: orchestratortest.MockServiceID,
+		CatalogId:      orchestratortest.MockCatalogID,
+		AssuranceLevel: &AssuranceLevelHigh,
+	})
+	assert.NoError(t, err)
+
+	// update the toe's assurance level and send the update request
+	toe, err = orchestratorService.UpdateTargetOfEvaluation(context.Background(), &orchestrator.UpdateTargetOfEvaluationRequest{
+		Toe: &orchestrator.TargetOfEvaluation{
+			CloudServiceId: orchestratortest.MockServiceID,
+			CatalogId:      orchestratortest.MockCatalogID,
+			AssuranceLevel: &AssuranceLevelMedium,
+		},
+	})
+	assert.NoError(t, err)
+	assert.NotNil(t, toe)
+	assert.Equal(t, "new assurance level", toe.AssuranceLevel)
+}
+
+func Test_RemoveTargetOfEvaluation(t *testing.T) {
```

Comment:
```suggestion
func TestService_RemoveTargetOfEvaluation(t *testing.T) {
```

Reply:
Are we naming Tests always "TestService" when the tested method is executed _on the service_? Then we should rename a few other test methods as well

Reply:
Tests should follow the syntax `TestMyGlobalFunc` if it is a global func that is being tested (`GlobalFunc`) or `TestMyStruct_MyFunc` if it is a method function of `MyStruct`.

This is defined in some style guide, I don't have a reference for it. Maybe @lebogg has it.